modules/internal/ChapelArray.chpl

```diff
@@ -2027,7 +2030,28 @@ module ChapelArray {
     proc tail(): this._value.eltType {
       return this[this.domain.high];
     }
-
+    pragma "no doc"
+    proc resizeAllocDomain(d: domain, d2: domain, factor=arrayAsVecGrowthFactor, direction=1, grow=1) {
+      const lo = d.low,
+            hi = d.high,
+            size = hi - lo + 1;
+      var newSize: int;
+      if grow > 0 {
+        newSize = max(size+1, (size*factor):int); // Always grow by at least 1.
```

Comment:
I read the FBVector article and used the 1.5 growth factor out of it.  I planned to leave the allocator specifics for another commit.  chpl_mem_good_alloc_size() is new enough that I hadn't thought to use it here though yet, so thanks for the tip.